File: app/src/main/java/com/example/mytemplate/config/GlobalConfig.java
package com.example.mytemplate.config;

public class GlobalConfig {

    private static final boolean isDebug = true;
    private static final String baseUrlProduction = "";
    private static final String baseUrlDev = "";
    private static final String baseUrlExample = "https://api.github.com/";
    private static final String sharePreferenceSession = "template_sp_session";

    public static String getBaseUrl(){
        return isDebug ? baseUrlDev : baseUrlProduction;
    }

    public static String getBaseUrlExample(){
        return baseUrlExample;
    }

    public static String getSharePreferenceSession() {
        return sharePreferenceSession;
    }
}

File: app/src/main/java/com/example/mytemplate/config/GlobalConfig.kt
package com.example.mytemplate.config

object GlobalConfig {
    const val isDebug = true

    // base url of application
    private const val baseUrlExample = "https://api.github.com/"
    private const val baseUrlProduction = baseUrlExample
    private const val baseUrlDev = baseUrlExample
    val baseUrl: String get() = if (isDebug) baseUrlDev else baseUrlProduction

    // shared preference
    const val sharePreferenceSession = "template_sp_session"
}